plugnpy/metric.py

```diff
@@ -115,7 +115,7 @@ def __init__(
         except (ValueError, TypeError) as ex:
             raise Exception("Invalid value for performance data precision '{0}': {1}".format(perf_data_precision, ex))
         try:
-            self.value = float(self.value)
+            value = float(self.value)
             self.perf_data = Metric.calculate_perf_data(name, value, unit, warning_threshold, critical_threshold,
```

Comment:
This line should be changed to self.value

Reply:
It was my first fix but then I changed as I thought the intent was to keep the instance attribute unmodified. FWIW, both fixes work.

I'll revert to my original and update the PR.

Reply:
Another alternative is to make `self.perf_data` a property and move the `Metric.calculate_perf_data` implementation there.

For now I've stuck with the plan but let me know if this small refactor is something you'd be amenable to.

Reply:
That's a valid point about leaving the instance variable unchanged. The rationale behind my suggestion was that I couldn't see a use case where leaving it as a string, if it could be cast to a float, would be useful. 

I really like the idea about making this into a property and don't see any disadvantages, but I would like to keep the implementation in Metric.calculate_perf_data just for the added flexibility of generating the perf_data string without a Metric instance.

Reply:
On further thought, it doesn't really make sense to make this into a property as it's only set once in the constructor, so I'll merge your current PR.